cosmic-applet-battery/src/backend/mod.rs

```diff
@@ -207,3 +210,55 @@ pub enum PowerProfileUpdate {
     Update { profile: Power },
     Error(String),
 }
+
+// check if battery charging thresholds is set
+pub async fn get_charging_limit() -> bool {
+    if let Ok(conn) = Connection::system().await {
+        if let Ok(backend) = get_power_backend(&conn, &BackendType::S76PowerDaemon).await {
+            match backend {
+                Backend::S76PowerDaemon(proxy) => {
+                    if let Ok((start, end)) = proxy.get_charge_thresholds().await {
+                        return start > 0 || end > 0;
+                    }
+                }
+                Backend::PowerProfilesDaemon(_) => {
+                    tracing::info!("Power Profiles Daemon is not supported.");
+
+                    return false;
+                }
+            };
+        }
+    }
+    false
+}
+
+// set battery charging thresholds via s76 power_daemon
+pub fn set_charging_limit(limit: bool) -> Result<bool> {
+    let rt = Runtime::new()?;
+
+    let success = rt.block_on(async {
```

Comment:
We shouldn't block the runtime. This should either be spawned in the background with `tokio::spawn`, or returned as a `Command`

Reply:
Ok, got it! The Command sounds good :)